### PR TITLE
Added is_structured property to BaseData and CSVData

### PR DESCRIPTION
**Pull request by ChrisWallace2020**
Boolean property that returns if a data object represents structured data. Raises NotImplemented for BaseData and checks that the selected data format is data frame for the CSV. Will be added to all Data readers in subsequent PRs.

### CODE REVIEW

File: dataprofiler/data_readers/csv_data.py [hunk 1]
```diff
@@ -106,6 +106,10 @@ def quotechar(self):
     def header(self):
         return self._header
 
+    @property
+    def is_structured(self):
+        return self.data_format == "dataframe"
```
Comment (gme5078) [inline dataprofiler/data_readers/csv_data.py:111]: I think this should just always be true? Isnt record format also structured?

File: dataprofiler/data_readers/base_data.py [hunk 2]
```diff
@@ -72,6 +72,10 @@ def data(self):
     def data_format(self):
         return self._selected_data_format
 
+    @property
+    def is_structured(self):
```
Comment (JGSweets) [inline dataprofiler/data_readers/base_data.py:76]: let's add a comment describing the property. I know others may not, but we should add those in another task and make sure to add in future.

File: dataprofiler/data_readers/csv_data.py [hunk 3]
```diff
@@ -106,6 +106,10 @@ def quotechar(self):
     def header(self):
         return self._header
 
+    @property
+    def is_structured(self):
```
Comment (JGSweets) [inline dataprofiler/data_readers/csv_data.py:110]: let's add a comment describing the property. I know others may not, but we should add those in another task and make sure to add in future.

File: dataprofiler/data_readers/base_data.py (hunk 2 above)
Comment (lettergram) [reply dataprofiler/data_readers/base_data.py:76]: Is this necessary?

File: dataprofiler/data_readers/csv_data.py (hunk 1 above)
Comment (lettergram) [reply dataprofiler/data_readers/csv_data.py:111]: ^ Yes

File: dataprofiler/tests/data_readers/test_csv_data.py [hunk 4]
```diff
@@ -376,6 +376,24 @@ def test_len_data(self):
...
+
+        # With option specifying dataframe as data_format
+        data = CSVData(input_file_path=filename,
+                       options={"data_format": "dataframe"})
+        self.assertTrue(data.is_structured)
+
+        # With option specifying records as data_format
+        data = CSVData(input_file_path=filename,
+                       options={"data_format": "records"})
+        self.assertFalse(data.is_structured)
```
Comment (lettergram) [inline dataprofiler/tests/data_readers/test_csv_data.py:396]: Wouldn't we need tests for other stuff?

File: dataprofiler/data_readers/csv_data.py (hunk 1 above)
Comment (ChrisWallace2020) [reply dataprofiler/data_readers/csv_data.py:111]: @JGSweets can you confirm? I thought I remember you saying it was dependent on data format and that records were a way to strip away structure?

File: dataprofiler/tests/data_readers/test_csv_data.py (hunk 4 above)
Comment (ChrisWallace2020) [reply dataprofiler/tests/data_readers/test_csv_data.py:396]: Like what? If its just dependent on data format and I checked the 2 data formats what else can I confirm?

File: dataprofiler/data_readers/csv_data.py (hunk 1 above)
Comment (JGSweets) [reply dataprofiler/data_readers/csv_data.py:111]: @ChrisWallace2020 is correct. Records is a list of strings, basically treating it as a text file.

File: dataprofiler/data_readers/base_data.py (hunk 2 above)
Comment (JGSweets) [reply dataprofiler/data_readers/base_data.py:76]: I believe it should clarify the intent, no?

File: dataprofiler/data_readers/csv_data.py (hunk 1 above)
Comment (JGSweets) [reply dataprofiler/data_readers/csv_data.py:111]: The intent is for the profiler to know how to profile the dataset. Records would not be able to be profiled by the structured profiler

File: dataprofiler/data_readers/base_data.py (hunk 2 above)
Comment (JGSweets) [reply dataprofiler/data_readers/base_data.py:76]: In terms of if it should be in base? Yes, I believe all our readers should clarify hence it should be in the abstract so anyone deriving from the base would know to include it.

**Review by JGSweets — APPROVED** [review]

File: dataprofiler/tests/data_readers/test_csv_data.py (hunk 4 above)
Comment (lettergram) [reply dataprofiler/tests/data_readers/test_csv_data.py:396]: I think you need to test all the expected data types: parquet, avro, csv, json

**Review by gme5078 — APPROVED** [review]

File: dataprofiler/tests/data_readers/test_csv_data.py (hunk 4 above)
Comment (JGSweets) [reply dataprofiler/tests/data_readers/test_csv_data.py:396]: I broke out each data type into its own task

**Pull request merged**